bug(developer): kmc bundling failing due to es5 target environment

Per https://build.palaso.org/buildConfiguration/Keyman_Developer_Release/372493?buildTab=log&focusLine=10471&logView=flowAware&linesState=153:

```
01:30:21   > bundle-kmc
01:30:21   > esbuild build/src/kmc.js --bundle --platform=node > build/cjs-src/kmc.cjs
01:30:21   
<snip>
01:30:22   
01:30:22   X [ERROR] Transforming class syntax to the configured target environment ("es5") is not supported yet
01:30:22   
01:30:22       ../../../common/web/keyman-version/keyman-version.mts:3:19:
01:30:22         3 │     export default class KEYMAN_VERSION {
01:30:22           ╵                    ~~~~~
01:30:22   
01:30:22     The target environment was set to "es5" here:
01:30:22   
01:30:22       ../../../common/web/keyman-version/tsconfig.json:11:14:
01:30:22         11 │     "target": "es5",
01:30:22            ╵               ~~~~~
01:30:22   
01:30:22   1 warning and 1 error
01:30:22   node:child_process:867
01:30:22       throw err;
01:30:22       ^
01:30:22   
01:30:22   Error: Command failed: C:\BuildAgent\work\deb663ed6681f452\keyman\node_modules\esbuild\lib\downloaded-esbuild-windows-64-esbuild.exe build/src/kmc.js --bundle --platform=node
01:30:22       at checkExecSyncError (node:child_process:826:11)
01:30:22       at Object.execFileSync (node:child_process:864:15)
01:30:22       at Object.<anonymous> (C:\BuildAgent\work\deb663ed6681f452\keyman\node_modules\esbuild\bin\esbuild:211:28)
01:30:22       at Module._compile (node:internal/modules/cjs/loader:1101:14)
01:30:22       at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
01:30:22       at Module.load (node:internal/modules/cjs/loader:981:32)
01:30:22       at Function.Module._load (node:internal/modules/cjs/loader:822:12)
01:30:22       at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
01:30:22       at node:internal/main/run_main_module:17:47 {
01:30:22     status: 1,
01:30:22     signal: null,
01:30:22     output: [ null, null, null ],
01:30:22     pid: 116992,
01:30:22     stdout: null,
01:30:22     stderr: null
01:30:22   }
01:30:22   [developer/src/kmc] ## bundle failed
```

This may be a regression emerging from #8429?

--- a/developer/src/kmc/package.json
+++ b/developer/src/kmc/package.json
@@ -12,10 +12,10 @@
   "scripts": {
     "build": "tsc -b",
     "bundle": "npm run bundle-kmc && npm run bundle-kmlmc && npm run bundle-kmlmi && npm run bundle-kmlmp",
-    "bundle-kmc": "esbuild build/src/kmc.js --bundle --platform=node > build/cjs-src/kmc.cjs",
-    "bundle-kmlmc": "esbuild build/src/kmlmc.js --bundle --platform=node > build/cjs-src/kmlmc.cjs",
-    "bundle-kmlmi": "esbuild build/src/kmlmi.js --bundle --platform=node > build/cjs-src/kmlmi.cjs",
-    "bundle-kmlmp": "esbuild build/src/kmlmp.js --bundle --platform=node > build/cjs-src/kmlmp.cjs",
+    "bundle-kmc": "esbuild build/src/kmc.js --bundle --platform=node --target=es2022 > build/cjs-src/kmc.cjs",
+    "bundle-kmlmc": "esbuild build/src/kmlmc.js --bundle --platform=node --target=es2022 > build/cjs-src/kmlmc.cjs",
+    "bundle-kmlmi": "esbuild build/src/kmlmi.js --bundle --platform=node --target=es2022 > build/cjs-src/kmlmi.cjs",
+    "bundle-kmlmp": "esbuild build/src/kmlmp.js --bundle --platform=node --target=es2022 > build/cjs-src/kmlmp.cjs",
     "test": "cd test && tsc -b && cd .. && c8 --reporter=lcov --reporter=text mocha",
     "prepublishOnly": "npm run build"
   },

